lib/posthog/client.rb

```diff
@@ -139,58 +142,56 @@ def queued_messages
       @queue.length
     end
 
-    def is_feature_enabled(flag_key, distinct_id, default_value = false, groups = {})
-      unless @personal_api_key
-        logger.error(
-          'You need to specify a personal_api_key to use feature flags'
-        )
-        return
-      end
-      is_enabled =
-        @feature_flags_poller.is_feature_enabled(
-          flag_key,
-          distinct_id,
-          default_value,
-          groups
-        )
-      capture(
-        {
-          'distinct_id': distinct_id,
-          'event': '$feature_flag_called',
-          'properties': {
-            '$feature_flag': flag_key,
-            '$feature_flag_response': is_enabled
-          }
-        }
-      )
-      return is_enabled
+    def is_feature_enabled(flag_key, distinct_id, default_value = false, groups: {}, person_properties: {}, group_properties: {})
+      return !!get_feature_flag(flag_key, distinct_id, default_value, groups: groups, person_properties: person_properties, group_properties: group_properties)
     end
 
-    def get_feature_flag(key, distinct_id, groups={})
-      unless @personal_api_key
-        logger.error(
-          'You need to specify a personal_api_key to use feature flags'
+    # Returns whether the given feature flag is enabled for the given user or not
+    #
+    # @param [String] key The key of the feature flag
+    # @param [String] distinct_id The distinct id of the user
```

Comment:
Structural question: For groups, passing distinct_id might not make sense, _especially_ on the backend side where a user might not even exist.